feat: Logging docs for Dart and Flutter (#4799)
getsentry/sentry-docs: src/platforms/dart/configuration/integrations/logging.mdx

---
title: Logging Integration
caseStyle: canonical
supportLevel: production
sidebar_order: 3
---

The `sentry_logging` library provides [Logging](https://pub.dev/packages/logging) support for Sentry via [onRecord property](https://pub.dev/documentation/logging/latest/logging/Logger/onRecord.html) that sends events and breadcrumbs to Sentry. Once this integration is configured you can use Logging’s public API.

The source can be found [on GitHub](https://github.com/getsentry/sentry-dart/tree/main/logging/lib/src).

## Install

> To add the Logging integration, add the `sentry-android-sentry_logging` dependency.

Reviewer: ```suggestion
To add the Logging integration, add the `sentry_logging` dependency.
```

```yml {filename:pubspec.yaml}

> dependencies:
>   sentry: ^6.3.0
>   sentry_logging: ^6.3.0

Reviewer: ```suggestion
dependencies:
  sentry: ^6.3.0
  sentry_logging: ^6.3.0
  logging: ^1.0.2
```

---
title: Logging Integration
caseStyle: canonical
supportLevel: production

> sidebar_order: 3

Author: Thank you, I believe none of the Android integrations have the `description`, I believe we've copied and pasted the very same "template".

---

> The `sentry_logging` library provides [Logging](https://pub.dev/packages/logging) support for Sentry via [onRecord property](https://pub.dev/documentation/logging/latest/logging/Logger/onRecord.html) that sends events and breadcrumbs to Sentry. Once this integration is configured you can use Logging’s public API.

Reviewer 2: ```suggestion
The `sentry_logging` library provides [Logging](https://pub.dev/packages/logging) support for Sentry via [onRecord property](https://pub.dev/documentation/logging/latest/logging/Logger/onRecord.html). It is able to collect breadcrumbs and capture events. Once this integration is configured you can use Logging’s public API exclusively or in combination to the Sentry's SDK API to capture and enrich events.
```

Reviewer 3: IMO, this update seems better in that it provides more information. If you two decide to keep it and the content applies to Timber, then I'd make the update there as well. The edits I made before still apply (avoiding Latin words and the same missing comma).

Author: Thanks, I will merge both suggestions.

> The `sentry_logging` library provides [Logging](https://pub.dev/packages/logging) support for Sentry via [onRecord property](https://pub.dev/documentation/logging/latest/logging/Logger/onRecord.html) that sends events and breadcrumbs to Sentry. Once this integration is configured you can use Logging’s public API.

Reviewer 3: ```suggestion
The `sentry_logging` library provides [Logging](https://pub.dev/packages/logging) support for Sentry using the  [onRecord property](https://pub.dev/documentation/logging/latest/logging/Logger/onRecord.html) that sends events and breadcrumbs to Sentry. Once this integration is configured, you can use Logging’s public API.
```

> The source can be found [on GitHub](https://github.com/getsentry/sentry-dart/tree/main/logging/lib/src).

Reviewer 2: What about pointing to the root of the package so users land on the readme and see the test/src projects?

```suggestion
The source can be found [on GitHub](https://github.com/getsentry/sentry-dart/tree/main/logging/).
```

> The source can be found [on GitHub](https://github.com/getsentry/sentry-dart/tree/main/logging/lib/src).

Reviewer 3: ```suggestion
You can find the source on [GitHub](https://github.com/getsentry/sentry-dart/tree/main/logging/lib/src).
```

Author: Similar to https://github.com/getsentry/sentry-docs/pull/4799#discussion_r820745510
@imatwawana do you have a preference?

Reviewer 3: I don't have a strong preference. Generally, it's good to use active instead of passive voice, but there's an argument for using passive voice here, so you can leave it.

## Install

To add the Logging integration, add the `sentry_logging` dependency.

```yml {filename:pubspec.yaml}
dependencies:

>   sentry: ^6.3.0
>   sentry_logging: ^6.3.0
>   logging: ^1.0.2

Reviewer 2: is the package on release registry?

```suggestion
  sentry: ^{{ packages.version('sentry.dart') }}
  sentry_logging: ^{{ packages.version('sentry.logging') }}
  logging: ^1.0.2
```

Author: Not yet, that's why it's not using `packages.version` otherwise versions won't be aligned, I will fix that once I add it to the release registry.

Author: https://github.com/getsentry/sentry-release-registry/pull/66
